state/secrets.go

```diff
@@ -618,7 +638,9 @@ func (st *State) deleteSecrets(uris []*secrets.URI, revisions ...int) (external
 					return nil, errors.Annotatef(err, "updating backend refcounts for %s", uri.String())
 				}
 			}
-			return nil, nil
+			// ???? Why we donnot remove the external resources from backend if we are not removing all revisions?????
+			// return nil, nil
+			return external, nil
```

Comment:
Revert back to nil. Because the uniter removes secret resources from the backend before removing them from the state. To keep consistency, we now do the same way for user secret revision removal.